src/br/com/zup/nossacasacodigo/cart/Cart.java

```diff
@@ -0,0 +1,46 @@
+package br.com.zup.nossacasacodigo.cart;
+
+import java.math.BigDecimal;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+import br.com.zup.nossacasacodigo.book.Book;
+
+public class Cart {
+	private Set<CartItem> items = new HashSet<>();
+	private static BigDecimal finalValue = BigDecimal.ZERO;
+
+	public void addToCart(Book book, int quantity) {
+		Optional<CartItem> alreadyInCart = checkIfIsInCart(book);
+		if (alreadyInCart.isPresent()) {
+			CartItem item = alreadyInCart.get();
+			item.setQuantity(item.getQuantity()+ quantity);
+		} else {
+			CartItem newItem = new CartItem(book, quantity);
+			items.add(newItem);
+		}
+	
+		finalValue = finalValue.add(book.getPrice().multiply(new BigDecimal(quantity)));
+	}
+
+	private Optional<CartItem> checkIfIsInCart(Book book) {
+		for (CartItem cartItem : items) {
+			if (cartItem.getBook() == book) {
```

Comment:
O `==` checa sempre a referência do livro. O que acontece se o livro for igual, mesmo nome, isbn, etc, mas estiver em referências diferentes?